app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt

```diff
@@ -94,17 +99,25 @@ class SolendEarnPresenter(
         deposits = newDeposits
 
         when {
+            newDeposits.any { it.supplyInterest == null } -> {
+                showDepositsWidgetError()
+                view?.setRatesErrorVisibility(isVisible = true)
+            }
             newDeposits.any { it is SolendDepositToken.Active } -> {
                 val activeDeposits = newDeposits.filterIsInstance<SolendDepositToken.Active>()
                 val depositBalance = activeDeposits.sumOf { it.usdAmount }
-                val totalYearBalance = activeDeposits.sumOf { it.usdAmount / BigDecimal(100) * it.supplyInterest }
+                val totalYearBalance = activeDeposits.sumOf {
```

Comment:
are we sure that calculating stuff should be done in Presenter? :)

Reply:
where do you want to do this?

Reply:
it is calculation for widget ticker